File: Easy/0014_longest-common-prefix/longest-common-prefix.go
package main

import (
	"fmt"
)

func longestCommonPrefix(strs []string) string {
	minLen := len(strs[0])
	for _, s := range strs {
		if len(s) < minLen {
			minLen = len(s)
		}
	}

	for i := 0; i < minLen; i++{
		c := strs[0][i]
		for _, s := range strs {
			if c != s[i] {
				return strs[0][:i]
			}
		}
	}
	return strs[0][:minLen]
}

func main() {
	tests := [][]string{
				{"flower", "flow", "flight"},
        {"dog", "racecar", "car"},
        {"a"},
	}

	for _, t := range tests {
		fmt.Printf("%v => %s\n", t, longestCommonPrefix(t))
	}
}

File: Easy/0014_longest-common-prefix/longest-common-prefix.py
#!/bin/python3

from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix : str = ""
        for i in range(len(strs[0])):
            c = strs[0][i]
            for s in strs:
                if i >= len(s) or c != s[i]:
                    return prefix
            prefix += c
        return prefix


def main():
    tests: List[List[str]] = [
        ["flower", "flow", "flight"],
        ["dog", "racecar", "car"],
        ["a"]
    ]
    for t in tests:
        print(t, "=>", Solution().longestCommonPrefix(t))

if __name__ == "__main__":
    main()
